kazoo.go

```diff
@@ -219,6 +231,470 @@ func (kz *Kazoo) create(node string, value []byte, ephemeral bool) (err error) {
+	}
+	return kz.createOrUpdateTopicPartitionAssignmentPathInZK(topic, partitionReplicaList, map[string]interface{}{}, true)
+}
+
+func (kz *Kazoo) getManualReplicaAssignment(replicaAssignmentList string, availableBrokerList []int, startPartitionID int, checkBrokerAvailable bool) (partitionReplicaList map[int][]int, err error) {
+	partitionList := strings.Split(replicaAssignmentList, ",")
+	partitionReplicaList = make(map[int][]int)
+	partitionID := startPartitionID
+	partitionList = partitionList[partitionID:]
+	for _, partition := range partitionList {
+		brokerList := strings.Split(partition, ":")
+		brokerIDs := make([]int, 0, len(brokerList))
+		for _, broker := range brokerList {
+			var brokerID int
+			brokerID, err = strconv.Atoi(broker)
+			if err != nil {
+				return
+			}
+			brokerIDs = append(brokerIDs, brokerID)
+		}
+		if len(brokerIDs) <= 0 {
+			err = fmt.Errorf("replication factor must be larger than 0")
+			return
+		}
+		brokerSet := make(map[int]struct{}, len(brokerIDs))
+		for _, brokerID := range brokerIDs {
+			brokerSet[brokerID] = struct{}{}
+		}
+		if len(brokerSet) != len(brokerIDs) {
+			err = fmt.Errorf("duplicate brokers in replica assignment: %v", brokerList)
+			return
+		}
+		if checkBrokerAvailable {
+			for _, broker := range brokerIDs {
+				isAvailable := false
+				for _, avBroker := range availableBrokerList {
+					if avBroker == broker {
+						isAvailable = true
+					}
+				}
+				if !isAvailable {
+					err = fmt.Errorf("some specified brokers not available. unavailable brokers: %d", broker)
+					return
+				}
+			}
+		}
+		partitionReplicaList[partitionID] = brokerIDs
+		if len(partitionReplicaList[partitionID]) != len(partitionReplicaList[startPartitionID]) {
+			err = fmt.Errorf("partition %d  has different replication factor: %v", partitionID, brokerList)
+		}
+		partitionID++
+	}
+
+	return
+}
+
+type topicAndPartition struct {
+	Topic     string
+	Partition int
+}
+
+func (kz *Kazoo) getReplicaAssignmentForTopics(topics ...string) (existingPartitionsReplicaList map[topicAndPartition][]int, err error) {
+	existingPartitionsReplicaList = make(map[topicAndPartition][]int)
+	for _, topic := range topics {
+		var data []byte
+		path := kz.getTopicPath(topic)
+		data, _, err = kz.conn.Get(path)
+		if err != nil {
+			continue
+		}
+		var pathData map[string]interface{}
+		err = json.Unmarshal(data, &pathData)
+		if err != nil {
+			continue
+		}
+		partitionValue, ok := pathData["partitions"]
+		if !ok {
+			continue
+		}
+		replicaMap, ok := partitionValue.(map[string]interface{})
+		if !ok {
+			continue
+		}
+		for partition, replicas := range replicaMap {
+			var (
+				partitionID  int
+				replicasData []int
+			)
+			partitionID, err = strconv.Atoi(partition)
+			if err != nil {
+				return
+			}
+			rawReplicas, ok := replicas.([]interface{})
+			if !ok {
+				continue
+			}
+			for _, rep := range rawReplicas {
+				repId, ok := rep.(int)
+				if !ok {
```

Comment:
On my machine, it looks that "ok" is always "false" so that I have to comment out this if statement to make it works properly,  perhaps it is because of the go version? 
my go version: go1.7.1 darwin/amd64

Reply:
Same here. `rep` is actually a `float64`, but even after fixing that, doesn't seem to work: the ids I get don't allow indexing into the `brokers` list with a trivial 1 partition / 1 broker case (I get eg replica = `1001`, which is then used to try to index the array of brokers)